src/lib.rs

```diff
@@ -852,6 +852,14 @@ impl<'a> AuthorizationRequest<'a> {
         self
     }
 
+    ///
+    /// Overrides the `redirect_url` to the one specified.
+    /// 
+    pub fn set_redirect_url(mut self, redirect_url: &'a RedirectUrl) -> Self {
```

Comment:
I think this lifetime may make the API tricky to use since clients will have to ensure that the `RedirectUrl` outlives the `AuthorizationRequest`. instead, let's change the type of `self.redirect_url` to be a `Cow<a', RedirectUrl>` and then we can pass an owned `RedirectUrl` here.

in the common case where users don't call `set_redirect_url`, we'll still avoid any extra clones, while supporting an easier API for this new use case.

Reply:
I thought that you might say that. Will do